spec/unit/util/windows/registry_spec.rb

```diff
@@ -163,5 +163,77 @@ def expects_registry_value(array)
         end
       end
     end
+
+    context "when reading binary values" do
+      let (:hklm) { Win32::Registry::HKEY_LOCAL_MACHINE }
+      let (:puppet_key) { "SOFTWARE\\Puppet Labs"}
+      let (:subkey_name) { "PuppetRegistryTest" }
+      let (:guid) { SecureRandom.uuid }
+
+      before :each do
+        hklm.create("#{puppet_key}\\#{subkey_name}", Win32::Registry::KEY_ALL_ACCESS) do |reg|
+          reg.write("#{guid}", Win32::Registry::REG_BINARY, "abinarystring")
+        end
+      end
+
+      after(:each) do
+        hklm.open(puppet_key, Win32::Registry::KEY_ALL_ACCESS) do |reg|
+          subject.delete_key(reg, subkey_name)
+        end
+      end
+
+      it "should return binary values" do
+
+        hklm.open("#{puppet_key}\\#{subkey_name}", Win32::Registry::KEY_ALL_ACCESS) do |reg|
+          vals = subject.values(reg)
+
+          expect(vals).to have_key(guid)
+          written = vals[guid]
+          expect(written).to eq("abinarystring")
+        end
+
+      end
+    end
+
+    # TODO Not sure about this.  It seems a bit hacky....
+    context "when reading corrupt values" do
+      let (:hklm) { Win32::Registry::HKEY_LOCAL_MACHINE }
+      let (:puppet_key) { "SOFTWARE\\Puppet Labs"}
+      let (:subkey_name) { "PuppetRegistryTest" }
+      let (:guid) { SecureRandom.uuid }
+
+      before :each do
+        # Create a registry import file with a corrupt DWORD
```

Comment:
I tried using REG.EXE and Ruby API calls but they all guarded against invalid values.  Eventually found I could craft a .REG file to make a bad value.  Need to test on 2008 though